r-package/grf/R/get_scores.R

```diff
@@ -254,9 +254,13 @@ get_scores.multi_arm_causal_forest <- function(forest,
       "."
     ))
   }
+  IPW <- matrix(0, length(subset), nlevels(W.orig))
+  IPW[observed.treatment.idx] <- 1 / W.hat[observed.treatment.idx]
```

Comment:
It's not immediately clear to me what's being computed here -- a comment would be helpful. Also, shouldn't `IPW` and `W.hat` be indexed into as matrices on line 258?

Thanks a lot for catching this bug!!

Reply:
It's doing the same thing as [policytree](https://github.com/grf-labs/policytree/blob/master/r-package/policytree/R/multi_arm_causal_forest-scores.R#L52), just filling in a matrix with the elements we want, the finally killing the first dimension since we only care about contrast:

```
Browse[1]>   IPW <- matrix(0, length(subset), nlevels(W.orig))
Browse[1]>   IPW[observed.treatment.idx] <- 1 / W.hat[observed.treatment.idx]
Browse[1]>   head(W, 10)
 [1] B C C B C C B A B A
Levels: A B C
Browse[1]>   head(IPW, 10)
        [,1]     [,2]     [,3]
 [1,] 0.0000 3.241808 0.000000
 [2,] 0.0000 0.000000 2.886212
 [3,] 0.0000 0.000000 2.519844
 [4,] 0.0000 2.951984 0.000000
 [5,] 0.0000 0.000000 2.850196
 [6,] 0.0000 0.000000 3.285741
 [7,] 0.0000 2.780404 0.000000
 [8,] 2.9757 0.000000 0.000000
 [9,] 0.0000 3.033285 0.000000
[10,] 3.1430 0.000000 0.000000
Browse[1]>   control <- IPW[, 1] != 0
Browse[1]>   IPW[control, -1] <- -1 * IPW[control, 1]
Browse[1]>   head(IPW, 10)
        [,1]      [,2]      [,3]
 [1,] 0.0000  3.241808  0.000000
 [2,] 0.0000  0.000000  2.886212
 [3,] 0.0000  0.000000  2.519844
 [4,] 0.0000  2.951984  0.000000
 [5,] 0.0000  0.000000  2.850196
 [6,] 0.0000  0.000000  3.285741
 [7,] 0.0000  2.780404  0.000000
 [8,] 2.9757 -2.975700 -2.975700
 [9,] 0.0000  3.033285  0.000000
[10,] 3.1430 -3.143000 -3.143000
```

what would be a helpful comment?

Reply:
Got it, thanks! I guess all that happened is that the syntax `IPW[observed.treatment.idx]` threw me off, since IPW is a matrix. It would have been helpful to me to have a brief comment on what the syntax is doing.

Reply:
Got it, added a comment on that. The matrix[matrix.index] subsetting is quite convenient, here it is in action for policytree:

```
library(policytree)
library(grf)
n <- 500
p <- 10
X <- matrix(rnorm(n * p), n, p)
W <- as.factor(sample(c("A", "B", "C"), n, replace = TRUE))
Y <- X[, 1] + pmax(0, X[, 2]) * (W == "B") + X[, 3] * (W == "C") + runif(n)
forest <- multi_arm_causal_forest(X, Y, W)
scores <- double_robust_scores(forest)

tree <- policy_tree(X, scores, 2)
pp <- predict(tree, X)
mean(scores[cbind(1:n, pp)])
# [1] 1.084409
colMeans(scores)
# A         B         C 
# 0.5101488 0.7949517 0.3626650 
```